model/enduser/registry.yaml

```diff
@@ -0,0 +1,32 @@
+groups:
+  - id: registry.enduser
+    type: attribute_group
+    display_name: End User Attributes
+    brief: >
+      Describes information about the end user, which can be used as a subdomain of browser, client, or user domains.
+    attributes:
+      - id: enduser.id
```

Comment:
It seems confusing to set `enduser.id = "QdH5CAWJgqVT4rOr0qtumf"` and `enduser.authentication.id = "lmolkova"`, based on the 

https://github.com/open-telemetry/semantic-conventions/pull/1146/files#r1712997369 and https://github.com/open-telemetry/semantic-conventions/pull/1146/files#r1710187141

It'd be more clear if we called this one `enduser.tracking.id` or `enduser.anonymous.id` so that people would not put PII there.

Reply:
per earlier discussion, it seemed that `anonymous` was confusing to some. 
i'm good with `enduser.anonymous_id` or `enduser.tracking_id`. neither `tracking` nor `anonymous` a namespace, nesting here doesn't seem to follow the naming convention?

Reply:
do you mean that `enduser.tracking.id` would not follow the naming convention? Why?

Reply:
I like `enduser.tracking.id`

Reply:
some examples https://github.com/open-telemetry/semantic-conventions/blob/d5d2b9d2a27f5983a5930699e0d40d5aba5c06fe/schemas/1.19.0#L32

nesting is used for `.`. if not nesting, use `_`.

Reply:
I'd recommend `.`, check out https://github.com/open-telemetry/semantic-conventions/issues/1462#issue-2576611847

> we have some attributes that would benefit from renaming (e.g. `db.cosmosdb.client_id` should stay consistent with `messaging.client.id`).

Reply:
we can discuss it in the next SIG. It seems inconsistent naming patterns used.

Reply:
can you elaborate on what's inconsistent?

Reply:
Semconv call: 

- Default: use `.` everywhere
- `_` is for old things and exceptional cases
- we need to document this guidance

Reply:
> I like `enduser.tracking.id`

I'm not a fan of `tracking` as it doesn't really "add" anything to the definition of what this field is really used for... And can have mixed meanings about what you are "tracking" about the enduser. eg. Is this used to identify that the `enduser` and opted in to enable tracking of their requests for support reasons and this is the id of the tracking session?

vs `anonymous` which is a more descriptive definition of the meaning of this property.

You could also use something `enduser.uaid` (Unique Anonymous Id), to be descriptive

Reply:
we prefer to avoid acronyms https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/attribute-naming.md#name-abbreviation-guidelines unless they are well-recognized

But I have no objections against `anonymous` or any other descriptive term.

Reply:
Yeah, sorry I didn't mean to "add" to and extend the discussion was just trying to clarify what the real meaning of this attribute will be for.

My choice from these would be `anonymous` as it makes it clearer (at least to me) about this "should" be used for (as someone will always do something unexpected)

Reply:
And then based on the Sem-Conv meeting this morning, I would suggest that it should be `enduser.anonymous.id` so that the `anonymous` namespace could be extended (if necessary) in the future.

Reply:
I'm good updating the PR to use `anonymous`, let's just confirm with @jsuereth before merging b/c I thought he had a concern with that term but don't recall specifics (and could be misremembering completely)

Reply:
https://github.com/open-telemetry/semantic-conventions/issues/1104#issuecomment-2243437378 here is josh's concern. do i need to wait for next SIG to get it answered? @jsuereth can you take a look at @MSNev's comment above? 
are you ok with `enduser.anonymous.id`? I'm good with `enduser.tracking.id` too if we have consensus.

Reply:
Not that I really want to extend the discussion further, but based on @jsuereth comments and understanding from his perspective, I can see how `anonymous` could also be misunderstood and given his suggestions of

> "user.unknown_id", "anonymous_user.id", "user.unauth_id".

From these comments, I would suggest the `unauth_id` option, would identify the meaning better as it's "unknown" but rather than a random (anonymous) identifier for the current `enduser` which (from a RUM perspective) which persisted for some period of time. As its unlikely that `unauth` would represent some common (extended) namespace then either `_` or `.` would work.

It's also represents a possible "linkable" identifier, while on it's own it doesn't identify the user, but in conjunction with other attributes it (may) be possible to identify the `enduser` so from that perspective it's not "truely" anonymous, which I infer is Josh's point on the name.

Reply:
My concerns around anonymous still apply.  I don't think the id is really anonymous and can't be considered as such.  It's a tracking id, and should be considered sensitive

Reply:
how about `enduser.unauthenticated.id`?

Reply:
+ `enduser.pseudo.id`

Let's create a list of options and add pros/cons.

Reply:
@lmolkova @trisch-me  Please review and fill in anything that i may have missed.: 

`enduser.pseudo.id`
     - pros: it's clear to know this id is not authenticated
     - cons: it might lead to misinterpretation of this id, like it's not a real id, a testing id, a temporary id?
     
`enduser.tracking.id'
     - pros: it's clear that this id is used to track a particular user.
     - cons: `tracking` may raise privacy concerns, as it implies monitoring user behavior, which could lead to user distrust. it also lacks context of what exactly is being tracked (e.g. user actions, sessions, locations, etc)
 
`enduser.unauth.id`
     - pros: `unauth` is short
     - cons: `unauth` is ambiguous, as it can be `unauthenticated` or `unauthorized`. additionally, acronym is not a good naming practice and leads to more confusion.
 
`enduser.temp.id` or `enduser.transient.id`
     - pros: it suggests that this id is temporary and associated with user who has not been authenticated.
     - cons: it lacks context about the id is temporary for what context (e.g. session, authentication)
 
`enduser.unauthenticated.id`
     - pro: it's clear to indicate an authenticated user.
     - cons: it collides with `enduser.authentication.id`, which can be renamed to `enduser.authenticated.id`, then it would have been fine?  
 
`enduser.anonymous.id`
     - pros: it's clear that this id is anonymous.
     - cons: it can be confusing and lacks context. as long we have a clear documentation, this should be ok?

Reply:
what were the pros for `enduser.unauthenticated.id`?

Reply:
we have already discussed and agreed to have `enduser.authentication.id`. `authentication` as a sub-namespace under enduser. anything that is not under `authentication` is either unauthenticated or unauthorized or anonymous or some random id.